### ISSUE TITLE
Update action returns list out of order

### ISSUE THREAD

**Issue by Petesta** [dixieio/redux-json-api#73]
I was wondering if there is a way to specify the ordering on an array or have the resulting array not change its ordering. Let's say you're referring to the users of a company, on `updateEntity`. For example, if all users are already listed in order by `id` then when you call `updateEntity` lets say to change the name of the first user in the array, the resulting array will return out of order.

```javascript
// The structure and ordering of data before the call
[
  {
    id: 1,
    name: "John"
  },
  {
    id: 2,
    name: "Emily"
  },
  {
    id: 3,
    name: "Jordan"
  }
]

// After call to updateEntity

[
  {
    id: 2,
    name: "Emily"
  },
  {
    id: 3,
    name: "Jordan"
  },
  {
    id: 1,
    name: "John Doe"
  }
]
```

**Comment by DanBradbury** [comment]
👍 Running into a similar issue where I'm either required to sort every time a record is updated or make another fetch after clearing out state (check for `API_WILL_READ` action and clear the payload in question)

I'd be curious if changing the default behavior would be desirable (i.e preserving ordering w/ find and replace on default) or if it would be possible to add options to `updateEntity`. Maybe something like `updateEntity(user, { preserveOrder: true })`?

Is the behavior described by @Petesta desired?

**Comment by Nopzen** [comment]
Personaly i think the behaviour of scrambling the data is okay.

The issue is that diffrent people uses diffrent `id`'s witch can either be incremental `id` witch this pr #74 by @Petesta would solve, but the issue it dosen't solve the issue for people using other `id`types such as `UUID`. 

Im not closed for the idea personally, but we need to atleast send a flag to the state updater witch key should be sorted on.

Just to be aware, that this could also affect views where a user **DON'T** want to have them sorted by id, and then we need to add a sorting on something that we already sorted?

**Comment by Petesta** [comment]
@Nopzen It's not a case of sorting by `id` or any key for that matter. I was just trying to use that as an example. That may have been confusing. Here's what's currently going on.

```javascript
[
  {
    name: 'Jordan',
    title: 'engineer'
  },
  {
    name: 'John Doe',
    title: 'engineer'
  },
  {
    name: 'Emily Jane',
    title: 'engineer'
  }
]

// After call to updateEntity

[
  {
    name: 'Jordan',
    title: 'engineer'
  },
  {
    name: 'Emily Jane',
    title: 'engineer'
  },
  {
    name: 'Sir John Doe',
    title: 'engineer'
  }
]
```

If I have a collection and pass it into a function that is supposed to update only an element of the collection I'd expect that function to preserve the structure of the collection minus the update on the element, thus also preserving whatever order it had.

Here's what I think should happen to a collection passed into a function. And in our case `updateEntity`.

```javascript
[
  {
    name: 'Jordan',
    title: 'engineer'
  },
  {
    name: 'John Doe',
    title: 'engineer'
  },
  {
    name: 'Emily Jane',
    title: 'engineer'
  }
]

// After call to updateEntity

[
  {
    name: 'Jordan',
    title: 'engineer'
  },
  {
    name: 'Sir John Doe',
    title: 'engineer'
  },
  {
    name: 'Emily Jane',
    title: 'engineer'
  }
]
```

It's an implementation detail in the code from what I can tell.

**Comment by Nopzen** [comment]
Ahh Sorry, that is my bad @Petesta I've understood you wrong, What i then understand is that if you update key 2 in the array you expect it to be on the key 2 again, that would make sense yes. Sorry for not catching the initial idea. I'll look talk a talk with @egeriis and then get back to you or just look into the PR.

**Comment by Nopzen** [comment]
@Petesta I did have s quick talk to @egeriis about the suggestion, we don't see why we shouldn't do this, i will have a look into your PR this week, and then lets take it from there.

Thanks for the contribution to the tool 🎉

**Comment by Petesta** [comment]
@Nopzen No problem 😄.

**Comment by Nopzen** [comment]
@Petesta - Its been merged and shiped with version `1.5.3`

I hope that will help you, closing.
